git_machete/git_operations.py

```diff
@@ -552,16 +552,13 @@ def flush_caches(self) -> None:
         self.__remotes_cached = None
         self.__counterparts_for_fetching_cached = None
         self.__short_commit_sha_by_revision_cached = {}
-        self.__tree_sha_by_commit_sha_cached = None
         self.__commit_sha_by_revision_cached = None
         self.__committer_unix_timestamp_by_revision_cached = None
         self.__local_branches_cached = None
         self.__remote_branches_cached = None
         self.__initial_log_shas_cached = {}
```

Comment:
Okay! if it comes to `__{initial,remaining}_log_shas_cached`... in the current shape it indeed **must be flushed** (since the key in both dicts is a branch name... and a branch can always be repointed to another commit).

You can consider, however, changing the `branch name -> log hashes` mapping to `full commit hash -> log hashes`. This way this new mapping will be immutable no matter what. You'll need to look up `get_commit_sha_by_revision()` first, however.